specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1002 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "Supports filtering by usageDate.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/grainParameter"
+          },
+          {
+            "$ref": "#/parameters/savingsPlanOrderIdParameter"
+          },
+          {
+            "$ref": "#/parameters/savingsPlanIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/BenefitUtilizationSummariesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "benefitRecommendationsListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The list of benefit recommendations.",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/benefitRecommendationModel"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The link (URL) to the next page of results.",
+          "readOnly": true,
+          "format": "uri"
+        }
+      },
+      "description": "Result of listing benefit recommendations."
+    },
+    "benefitRecommendationModel": {
+      "type": "object",
+      "description": "benefit plan recommendation details.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BenefitResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "type": "object",
+          "$ref": "#/definitions/benefitRecommendationProperties"
+        }
+      }
+    },
+    "benefitRecommendationProperties": {
+      "type": "object",
+      "description": "The properties of the benefit plan recommendation.",
+      "discriminator": "scope",
+      "properties": {
+        "firstConsumptionDate": {
+          "type": "string",
+          "description": "The first usage date used for looking back for computing the recommendation.",
+          "readOnly": true
+        },
+        "lastConsumptionDate": {
+          "type": "string",
+          "description": "The last usage date used for looking back for computing the recommendation.",
+          "readOnly": true
+        },
```

Comment:
what date formats are accepted?

Reply:
Is it '"format": "datetime" or a custom format?

Reply:
The format is date time with a value similar to "2021-12-02T00:00:00Z"
If I add format as 'date-time', swagger/open API hub validation fails because of character 'Z' at the end. Hence keeping it as a string.

Reply:
I think I saw another format like dateTimeZone in CADL, I wonder what format that generates.

Reply:
Updated files to use format 'date-time'